client/src/App.js

```diff
@@ -1,24 +1,44 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+import Registration from "./pages/Registration";
+import Navbar from "./pages/Navbar";
+import Login from "./pages/Login";
+import { BrowserRouter, Router, Routes, Route } from "react-router-dom";
+import "bootstrap/dist/css/bootstrap.min.css";
+import Dashboard from "./components/user/Dashboard";
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+axios.defaults.baseURL = "http://127.0.0.1:8000/";
+axios.defaults.headers.post["Content-Type"] = "application/json";
+axios.defaults.headers.post["Accept"] = "application/json";
+axios.defaults.withCredentials = true;
+axios.interceptors.request.use(function (config) {
+  const token = localStorage.getItem("auth_token");
+  config.headers.Authorization = token ? `Bearer ${token}` : "";
+  return config;
+});
 
 function App() {
+  const [auth, setAuth] = useState();
+
+  useEffect(() => {
+    setAuth(localStorage.getItem("auth_token"));
+  }, [auth]);
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <BrowserRouter>
+      <Navbar />
+      <div className="App">
+        <Routes>
+          <Route path="/" element={auth ? <Dashboard /> : <Login />} />
```

Comment:
@Brucenigel you can just check in your useEffect if the user is authenticated instead of manually putting conditions in each route.